pdns/stubresolver.cc

```diff
@@ -30,6 +38,47 @@ bool resolversDefined()
   return true;
 }
 
+/*
+ * Parse /etc/resolv.conf and add those nameservers to s_resolversForStub
+ */
+static void parseLocalResolvConf()
+{
+  ifstream ifs(LOCAL_RESOLV_CONF_PATH);
+  struct stat st;
+  string line;
+
+  if(!ifs)
+    return;
+  if (stat(LOCAL_RESOLV_CONF_PATH, &st) == -1)
+    return ;
+  s_localResolvConfMtime = st.st_mtime;
+
+  pthread_mutex_lock(&s_resolversForStubLock);
```

Comment:
the lock.hh include file has 'resource acquisition is initialization' (RAII) wrappers for setting and releasing locks. These are much recommended as you can't forget to unlock - the destructor does that for you.